tutorial bcc python: `disksnoop.py` not working

Hello,

I'm following the [tutorial bcc python](https://github.com/iovisor/bcc/blob/master/docs/tutorial_bcc_python_developer.md#lesson-6-disksnooppy) and when I try to execute the disksnooppy example I get this error:
```
/virtual/main.c:20:37: error: incomplete definition of type 'struct request'
                bpf_trace_printk("%d %x %d\n", req->__data_len,
                                               ~~~^
include/linux/blkdev.h:24:8: note: forward declaration of 'struct request'
struct request;
       ^
/virtual/main.c:21:10: error: incomplete definition of type 'struct request'
                    req->cmd_flags, delta / 1000);
                    ~~~^
include/linux/blkdev.h:24:8: note: forward declaration of 'struct request'
struct request;
       ^
2 errors generated.
Traceback (most recent call last):
  File "/root/bcc/./examples/tracing/disksnoop.py", line 20, in <module>
    b = BPF(text="""
  File "/usr/lib/python3.10/site-packages/bcc/__init__.py", line 479, in __init__
    raise Exception("Failed to compile BPF module %s" % (src_file or "<text>"))
Exception: Failed to compile BPF module <text>
```

And the thing is, the only occurrence of `blkdev.h` on my computer is found in `/tmp/kheaders-5.16.1-arch1-1/include/linux/blkdev.h`.
And it effectively starts with a sequence of empty structs:
```C
# head -n 35 /tmp/kheaders-5.16.1-arch1-1/include/linux/blkdev.h
/* SPDX-License-Identifier: GPL-2.0 */
#ifndef _LINUX_BLKDEV_H
#define _LINUX_BLKDEV_H

#include <linux/sched.h>
#include <linux/genhd.h>
#include <linux/list.h>
#include <linux/llist.h>
#include <linux/minmax.h>
#include <linux/timer.h>
#include <linux/workqueue.h>
#include <linux/wait.h>
#include <linux/bio.h>
#include <linux/gfp.h>
#include <linux/rcupdate.h>
#include <linux/percpu-refcount.h>
#include <linux/blkzoned.h>
#include <linux/sbitmap.h>

struct module;
struct request_queue;
struct elevator_queue;
struct blk_trace;
struct request;
struct sg_io_hdr;
struct blkcg_gq;
struct blk_flush_queue;
struct kiocb;
struct pr_ops;
struct rq_qos;
struct blk_queue_stats;
struct blk_stat_callback;
struct blk_crypto_profile;
```

--- a/examples/tracing/disksnoop.py
+++ b/examples/tracing/disksnoop.py
@@ -19,7 +19,7 @@ REQ_WRITE = 1		# from include/linux/blk_types.h
 # load BPF program
 b = BPF(text="""
 #include <uapi/linux/ptrace.h>
-#include <linux/blkdev.h>
+#include <linux/blk-mq.h>
 
 BPF_HASH(start, struct request *);
 
@@ -46,7 +46,10 @@ void trace_completion(struct pt_regs *ctx, struct request *req) {
 if BPF.get_kprobe_functions(b'blk_start_request'):
         b.attach_kprobe(event="blk_start_request", fn_name="trace_start")
 b.attach_kprobe(event="blk_mq_start_request", fn_name="trace_start")
-b.attach_kprobe(event="blk_account_io_done", fn_name="trace_completion")
+if BPF.get_kprobe_functions(b'__blk_account_io_done'):
+    b.attach_kprobe(event="__blk_account_io_done", fn_name="trace_completion")
+else:
+    b.attach_kprobe(event="blk_account_io_done", fn_name="trace_completion")
 
 # header
 print("%-18s %-2s %-7s %8s" % ("TIME(s)", "T", "BYTES", "LAT(ms)"))

